src/main/java/hlf/java/rest/client/service/impl/TransactionFulfillmentImpl.java

```diff
@@ -83,7 +106,12 @@ public ResponseEntity<ClientResponseModel> writeTransactionToLedger(
       // cause
       // information to know if it was an IOException so that a retry can be
       // attempted.
-      if (e.getCause() instanceof IOException) {
+
+      if (e.getCause() instanceof IOException
+          || e.getMessage().contains(FABRIC_FAILED_ORDERER_ERROR)
+          || e.getMessage().contains(FABRIC_GRPC_CONNECTION_ERROR)
+          || e.getMessage().contains(FABRIC_CONNECTION_CREATE_ERROR)
+          || e.getMessage().contains(FABRIC_CHAINCODE_LAUNCH_ERROR)) {
```

Comment:
[nitpick] maybe you can move these error codes into a list (say for example fabricErrorStrings) and within this If block evaluate it like fabricErrorStrings.stream().anyMatch(e.getMessage()::contains)
to make it compact.

Effectively it will look
`if (e.getCause() instanceof IOException || fabricErrorStrings.stream().anyMatch(e.getMessage()::contains))`